File: TaskPlusPlus.Application/DTOs/Category/CategoryDto.cs
using TaskPlusPlus.Application.DTOs.Common;
using TaskPlusPlus.Domain.ValueObjects.Category;

namespace TaskPlusPlus.Application.DTOs.Category;
public class CategoryDto : BaseDto, ICategoryDto
{
    public string Name { get; set; } = null!;
    public bool IsFavorite { get;  set; }
    public bool IsImmutable { get; set; }
    public string ColorHex { get; set; } = null!;
    public string Icon { get; set; } = null!;
    public CategorySettings Settings { get; set; } = null!;
}

File: client/src/app/shared/models/category/CategoryDto.ts
export interface CategoryDto {
    id: number;
    name: string;
    isFavorite: boolean;
    isImmutable: boolean;
    colorHex: string;
    icon: string;
    settings: CategorySettings;
}

export interface CategorySettings {
    grouping: string;
    sorting: string;
    direction: boolean;
}

export const InboxId: number = 1;
export const NextActionsId: number = 2;
export const SomedayMaybeId: number = 6;
export const WaitingForId: number = 5;